odl/space/numpy_ntuples.py

```diff
@@ -1047,53 +1056,89 @@ def __repr__(self):
         return '{}({})'.format(class_name, inner_str)
 
     # Copy these to handle bug in ABCmeta
```

Comment:
How about we scrap ABCMeta for spaces? I don't think many people will ever implement a space, and those who do can probably deal with the "soft" abstract method implementation via `NotImplementedError`.